### PR TITLE
fix(proxy): fix XHR HTTP requests from the dev proxy

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Fix XHR requests when using the dev proxy and a MAAS with TLS enabled.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- [Enable TLS](https://github.com/canonical-web-and-design/app-tribe/issues/847) on your MAAS.
- Run `yarn start`.
- Check that you can log in and out.
- Go to Settings -> commissioning scripts and check that you can upload a script (example scripts here: https://maas.io/docs/commissioning-script-reference).
- Run `yarn ui` and repeat the tests for log in/out and uploading a script.
- Turn off TLS on your MAAS.
- With both `yarn start` and `yarn ui` test that you can log in/out and upload a script.

## Fixes

Fixes: canonical-web-and-design/app-tribe#886.
Fixes: #3184.

### CODE REVIEW

File: proxy/package.json [hunk 1]
```diff
@@ -11,14 +11,14 @@
     "serve-frontends": "concurrently \"yarn serve-ui\" \"yarn serve-legacy\"",
     "serve-base": "concurrently \"yarn serve-root\" \"yarn serve-proxy\"",
     "serve-base-wait": "wait-on http://0.0.0.0:8401 && wait-on http://0.0.0.0:8402 && yarn serve-base",
-    "serve-proxy": "node ./index.js",
+    "serve-proxy": "nodemon ./index.js",
```
Comment (huwshimi) [inline proxy/package.json:14]: This is a drive-by to aid in development so that the proxy is reloaded when the express files are changed. This also watches for changes to .env.local so you can now change MAASes without restarting the proxy! I'm not sure why we haven't done this before...

File: proxy/scripts/ui.js [hunk 2]
```diff
@@ -7,7 +7,5 @@ console.log("");
 console.log("Wait for the React dev server to start and then visit:".red, `http://${address.ip()}:8400`.blue);
 console.log("The URL displayed by the React dev server message will be incorrect.".red);
 console.log("");
-console.log("Warning:".yellow, "This is the React client only. The legacy client will be unavailable and there may be some CSS differences.");
```
Comment (huwshimi) [inline proxy/scripts/ui.js:10]: This was resolved a while ago.

**Review by Caleb-Ellis — APPROVED** [review]
Nice - I knew there'd be some (not-so) hidden bugs caused by enabling native TLS. Thanks for fixing it!
